jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/YenKShortestPaths.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 2017, by Keve Müller.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.shortestpath;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.interfaces.KShortestPathAlgorithm;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm;
+
+/**
+ * Yen's algorithm computes single-source K-shortest loopless paths for a graph with non-negative
+ * edge cost. The algorithm was published by Jin Y. Yen in 1971 and employs any shortest path
+ * algorithm to find the best path, then proceeds to find K-1 deviations of the best path.
+ * <p>
+ * Based on pseudocode published at https://en.wikipedia.org/wiki/Yen%27s_algorithm

```

Comment:
Please state the runtime complexity of your implementation using big-O notation:
$O(runtime)$
Use the $$ signs.
See "javadoc and comment" and "math in javadoc": https://github.com/jgrapht/jgrapht/wiki/Contributor-Guidelines